Docs - Adds Local and Cloud dev workflows docs, create Workflow section of TOC (#838)
spatialos/UnrealGDK: SpatialGDK/Documentation/content/cloud-dev-workflow.md

# Cloud Development Workflow

> <!-- This is a live embed of a google drawing -->

Author: yep in develop drive and with big warning

> <!-- This is a live embed of a google drawing -->

Author: https://drive.google.com/drive/folders/1FugEDuIHe6Dy_ZSi-FawEumHSgtcbDoZ

<img src="https://docs.google.com/drawings/d/e/2PACX-1vQRmK7TxLji8pT7erPl54hqMMMDsdosZY1OZ2wuPYLQ23dWIrx86qCHggEeq-XasTCsqRe40fCKQvKN/pub?w=758&amp;h=1162">

You may find the below snippets useful as reference:

### Build server-worker assembly

```
Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat <YourProject>Server Linux Development <YourProject>.uproject
```

> Replacing **`<YourProject>`** with the name of your Unreal project.

Reviewer: ```suggestion
Replacing `<YourProject>` with the name of your Unreal project.
```
No need for bold when we reference file names, code, command line arguments. Bold for UI references.

### Build client-worker assembly

```
Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat <YourProject> Win64 Development <YourProject>.uproject
```

> Replacing **`<YourProject>`** with the name of your Unreal project.

Reviewer: ```suggestion
Replacing `<YourProject>` with the name of your Unreal project.
```

### Upload Assembly

```
spatial cloud upload myassembly
```

Replacing `myassembly` with the name you choose to give your assembly.

### Launch Cloud Deployment

```

> spatial cloud launch --snapshot=snapshots/default.snapshot <assembly_name> launch_config.json <deployment_name>

Reviewer: ```suggestion
spatial cloud launch --snapshot=snapshots/default.snapshot <myassembly> launch_config.json <deployment_name>
```
Use the term consistent with above.

```

Providing:

> * `<assembly_name>`, which identifies the worker assemblies to use (as chosen in the `spatial cloud upload` command).

Reviewer: Who chooses it? Our user? In which case:
```suggestion
* `<myassembly>` -  identifies the worker assemblies to use (as you specified earlier in the `spatial cloud upload` command). Replace `myassembly` with the name you choose to give your assembly.
```
Avoiding "which" means less text to read - nice in a list!  :-)
The explanation says "which assemblies to use" - does that mean you can specify more than one in the command? If not, we should say "assembly to use". If yes, we should say this explicitly.

> * `launch_config.json`, which declares the world and load balancing configuration.

Reviewer: ```suggestion
* `launch_config.json` - declares the world and load balancing configuration.
```
Avoiding "which" means less text to read - nice in a list! :-)

> * `<deployment name>`, which labels the deployment in the [Console](https://console.improbable.io).

Reviewer: ```suggestion
* `<deployment name>`- labels the deployment for SpatialOS to reference in the [Console]({{urlroot}}/content/glossary#console).
```
Better in this circumstance to link to the glossary entry as it gives a little explanation on what the Console is before linking to it. A sort of documentation amuse-bouche.

Author: good point, incorporated
